src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -375,6 +383,41 @@ public IOpenApiReferenceable ResolveReference(OpenApiReference reference)
             return ResolveReference(reference, false);
         }
 
+        /// <summary>
+        /// Takes in an OpenApi document instance and generates its hash value 
+        /// </summary>
+        /// <param name="doc">The OpenAPI description to hash.</param>
+        /// <returns>The hash value.</returns>
+        public static string GenerateHashValue(OpenApiDocument doc)
+        {
+            HashAlgorithm sha = SHA512.Create();
+            using var memoryStream = new MemoryStream();
+
+            using var cryptoStream = new CryptoStream(memoryStream, sha, CryptoStreamMode.Write);
+            using var streamWriter = new StreamWriter(cryptoStream);
+
+            var openApiJsonWriter = new OpenApiJsonWriter(streamWriter, new OpenApiJsonWriterSettings { Terse = true });
+            doc.SerializeAsV3(openApiJsonWriter);
+            openApiJsonWriter.Flush();
+
+            var hash = memoryStream.ToArray();
```

Comment:
```suggestion
            cryptoStream.FlushFinalBlock();
            var hash = sha.Hash;
```

according to https://stackoverflow.com/questions/4725507/calculate-hash-when-writing-to-stream

Reply:
Noted.